### PR TITLE
Migration tracking + appstore setup

### PR DESCRIPTION
**Pull request by ecotopian**
- track migration errors and performance
- track browser errors
- deploy to beta and live testflight in parallel

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -108,6 +156,13 @@ workflows:
               only: /^v.*/
             branches:
               ignore: /.*/
+      - build-testflight-deploy-beta:
```
Comment (vauxhall) [inline .circleci/config.yml:159]: ⚡

File: Client/Configuration/EcosiaBeta.xcconfig [hunk 2]
```diff
@@ -7,6 +7,6 @@
 
 MOZ_BUNDLE_DISPLAY_NAME = Ecosia Beta
 MOZ_BUNDLE_ID = com.ecosia.ecosiaapp.firefox
-CODE_SIGN_ENTITLEMENTS = Client/Entitlements/Ecosia.entitlements
+CODE_SIGN_ENTITLEMENTS = Client/Entitlements/EcosiaBeta.entitlements
```
Comment (vauxhall) [inline Client/Configuration/EcosiaBeta.xcconfig:10]: Why the `beta` and not just Ecosia?

File: Client/Ecosia/Migration/EcosiaImport.swift [hunk 3]
```diff
@@ -8,6 +8,10 @@ import MozillaAppServices
 import Storage
 import Shared
 
+private let tabsError = 911
```
Comment (vauxhall) [inline Client/Ecosia/Migration/EcosiaImport.swift:11]: Wondering what is the reasoning behind the numbers, why starting in `911` for example?

Would it make sense to have instead an `enum` of errors with a rawValue of `Int`?

File: Client/Ecosia/Migration/EcosiaImport.swift [hunk 4]
```diff
@@ -22,6 +26,11 @@ final class EcosiaImport {
 
     struct Failure: Error {
         let reasons: [MaybeErrorType]
+
+        var description: String {
+            // max 3 errors to be reported to save bandwidth and storage
```
Comment (vauxhall) [inline Client/Ecosia/Migration/EcosiaImport.swift:31]: Just an idea, but if bandwidth and storage are an issue (not sure they are), we could avoid sending error messages and only sending codes, as the messages might be the ones stressing our resources more?

**Review by vauxhall — APPROVED** [review]
Great work 👍 , a few comments but none of them are stoppers

File: Client/Configuration/EcosiaBeta.xcconfig (hunk 2 above)
Comment (ecotopian) [reply Client/Configuration/EcosiaBeta.xcconfig:10]: the beta has different app groups and keychain groups and therefore needs different entitlements. To give you some more context:
EcosiaBeta is the one that we used so far in TestFlight and installs in parallel to the "old" app (com.ecosia.ecosiaapp.firefox)
Ecosia has the same bundle id like the old app and will therefore overinstall it and trigger migration if installed

So far we'll deploy both in parallel to keep existing Beta testers. Once we fully switch we can remove the Beta configs again

File: Client/Ecosia/Migration/EcosiaImport.swift (hunk 3 above)
Comment (ecotopian) [reply Client/Ecosia/Migration/EcosiaImport.swift:11]: :D yeah that's just arbitrary: like 911 call for ambulance.. I did not come up with a better idea and I don't know if there are dedicated error codes for failing migrations. Any suggestions? Also those are "only" temporary as well and can be removed after all users have migrated.. sure an enum makes sense -> will give that a shot

File: Client/Configuration/EcosiaBeta.xcconfig (hunk 2 above)
Comment (vauxhall) [reply Client/Configuration/EcosiaBeta.xcconfig:10]: Makes sense, thanks for the brief 👍

File: Client/Ecosia/Migration/EcosiaImport.swift (hunk 4 above)
Comment (ecotopian) [reply Client/Ecosia/Migration/EcosiaImport.swift:31]: I am actually interested in the messages to know why exactly migrations are failing (.e.g. no disk space, corrupt files etc). The codes only tell which part (tabs, favs, history) fails. As those error messages can be long I decided together with Arnaud to cap them to 3 and also remove them from the snowplow database regularly. Also I expect not many errors but you can never know 🤞

File: Client/Ecosia/Migration/EcosiaImport.swift (hunk 3 above)
Comment (vauxhall) [reply Client/Ecosia/Migration/EcosiaImport.swift:11]: I'm fine with starting on 911, just thought it had a meaning. I would use an enum and let it decide the numbers by default, but any solution works for me 👍

File: Client/Ecosia/Migration/EcosiaImport.swift (hunk 4 above)
Comment (vauxhall) [reply Client/Ecosia/Migration/EcosiaImport.swift:31]: My idea was mostly having codes for tabs, favs, history, and also codes for the reasons: no disk space = 1, etc., makes them easier to aggregate also. But I'm fine with having messages too.

Comment (ecotopian) [reply Client/Ecosia/Migration/EcosiaImport.swift:33]: once we know through golden master rollout which errors can occur we can refine the tracking to add more specific codes - for now I don't which errors can actually happen - let's see

**Pull request merged**